WalletWasabi/Logging/Logger.cs

```diff
@@ -144,21 +139,27 @@ private static void Log(LogLevel level, string message, string category, int add
 				var messageBuilder = new StringBuilder();
 				messageBuilder.Append($"{DateTime.UtcNow.ToLocalTime():yyyy-MM-dd HH:mm:ss} {level.ToString().ToUpperInvariant()}");
 
-				if (message != "" && category != "") // If none of them empty.
-				{
-					messageBuilder.Append($" {category}: {message}{EntrySeparator}");
-				}
-				else if (message == "" && category != "")  // If only the message is empty.
+				if (message == "")
 				{
-					messageBuilder.Append($" {category}{EntrySeparator}");
-				}
-				else if (message != "" && category == "") // If only the category is empty.
-				{
-					messageBuilder.Append($": {message}{EntrySeparator}");
+					if (category == "") // If both empty. It probably never happens though.
+					{
+						messageBuilder.Append($"{EntrySeparator}");
+					}
+					else // If only the message is empty.
+					{
+						messageBuilder.Append($" {category}{EntrySeparator}");
+					}
 				}
-				else // if (message == "" && category == "") // If both empty. It probably never happens though.
+				else
 				{
-					messageBuilder.Append($"{EntrySeparator}");
+					if (category == "") // If only the category is empty.
+					{
+						messageBuilder.Append($": {message}{EntrySeparator}");
+					}
+					else // If none of them empty.
+					{
+						messageBuilder.Append($" {category}: {message}{EntrySeparator}");
+					}
```

Comment:
Neither this one.

Reply:
There is a small improvement here. We check `if (message == "")` once not twice and same for `if (message != "")`

Reply:
@nopara73 If message is not empty and category is empty (third case), we need to check message 3 times and category 3 times in order to get there, but with the proposed code we check each variable only one time.
I know it is not a lot but isn't this worth it?

Reply:
Yes, you are right. Sorry for closing it immaturely. I tried to cherry-pick your commit, but didn't work, so I just copied it.

https://github.com/zkSNACKs/WalletWasabi/pull/1700